src/libraries/component-time-scroll-view/TSVAnnotationLayer.tsx

```diff
@@ -0,0 +1,46 @@
+import { Annotation } from 'libraries/context-annotations';
+import { BaseCanvas } from 'libraries/figurl-canvas';
+import { convert1dDataSeries } from 'libraries/util-point-projection';
+import { Matrix } from 'mathjs';
+import { useCallback } from 'react';
+
+export type TSVAnnotationLayerProps = {
+    timeRange: [number, number]
+    margins: {left: number, right: number, top: number, bottom: number}
+    annotations: Annotation[]
+    timeToPixelMatrix: Matrix
+    width: number
+    height: number
+}
+
+const emptyDrawData = {}
+
+const TSVAnnotationLayer = (props: TSVAnnotationLayerProps) => {
+    const {width, height, margins, timeToPixelMatrix, annotations} = props
+    const paint = useCallback((context: CanvasRenderingContext2D) => {
+        context.clearRect(0, 0, context.canvas.width, context.canvas.height)
+    
+        const timepointAnnotationTimes = annotations.filter(a => (a.type === 'timepoint')).map(a => (a.timeSec))
+        const timepointAnnotationTimesPix = convert1dDataSeries(timepointAnnotationTimes, timeToPixelMatrix)
+    
+        for (let tp of timepointAnnotationTimesPix) {
+            context.strokeStyle = 'orange'
+            context.lineWidth = 2.5
+            context.beginPath()
+            context.moveTo(tp, margins.top)
+            context.lineTo(tp, context.canvas.height - margins.bottom)
+            context.stroke()
+        }
+    }, [margins, timeToPixelMatrix, annotations])
```

Comment:
done

Reply:
Thanks, great!
I think I'd go even further and pull the `paint` function out of the layer definition & just pass it the pixel annotations as its `drawData` rather than making that a callback dependency. Ultimately it'd be nice to think of a way to prompt anything using a `BaseCanvas` to redraw without actually going through an unnecessary rerender cycle, and being strictly functional about the paint functions would help make that possible, I'm thinking?

Reply:
This is an important conversation for us to have. I've been moving toward implementing paint as a callback and always using empty draw data. Let's chat about that sometime.